src/pyhf/cli/patchset.py

```diff
@@ -0,0 +1,161 @@
+"""The pyhf spec CLI subcommand."""
+import logging
+
+import click
+import json
+import sys
+import jsonpatch
+
+from ..workspace import Workspace
+from .. import utils
+
+logging.basicConfig()
+log = logging.getLogger(__name__)
+
+
+@click.group(name='patchset')
+def cli():
+    """Operations involving patchsets."""
+
+
+@cli.command()
+@click.argument('patchset', default='-')
+@click.option(
+    '--name', help='The name of the patch to extract.', default=None,
+)
+@click.option(
+    '--output-file',
+    help='The location of the output json file. If not specified, prints to screen.',
+    default=None,
+)
+@click.option(
+    '--with-metadata/--without-metadata',
+    default=False,
+    help="Include patchset metadata in output.",
+)
+def extract(patchset, name, output_file, with_metadata):
```

Comment:
I'm getting an error message at the moment when I try to use this function:

```bash
pyhf patchset extract --name C1N2_Wh_hbb_900_50 patchset.json
```

output:

```
Traceback (most recent call last):
  File "/anaconda3/bin/pyhf", line 8, in <module>
    sys.exit(cli())
  File "/anaconda3/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/anaconda3/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/anaconda3/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/anaconda3/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/anaconda3/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/pyhf/cli/patchset.py", line 52, in extract
    if patch['name'] == name:
KeyError: 'name'
```

Am I misusing the function somehow?

Reply:
`['metadata']['name']`